metabase/metabase
# FE: Database details permission

## Changes

<img width="1153" alt="Screenshot 2022-04-04 at 22 25 25" src="https://user-images.githubusercontent.com/14301985/161666865-b814baac-ecc9-4095-8d1c-3b84a6c077f2.png">

Adds the "Database details" permission to permission editor as well as enforcement for users. Needs BE to be merged https://github.com/metabase/metabase/pull/21531

## How to verify

- Open [Data Permissions page](http://localhost:3000/admin/permissions/data/) as an admin
- Create a Test User and login under it in another browser window

Mock user's permissions enterprise/frontend/src/metabase-enterprise/feature_level_permissions/utils.ts
Make `canAccessDatabaseManagement` to return `true`

- Open a database in the permission editor
- Change the `Manage database` permission for "All Users" for a database and save
    - Ensure it is updated
- Ensure the `Manage database` permission is visible only on a database level, not schema/table level
- Ensure Test User can go to Admin Settings and view the Database details page. It shows errors without BE.
